### PR TITLE
[TSS] update error

### PR DESCRIPTION
**Pull request by nkitlabs**
Fixed: 
- Update error on tss and bandtss module

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Review by taobun — APPROVED** [review]
lgtm

File: x/bandtss/keeper/keeper_signing.go [hunk 1]
```diff
@@ -129,17 +129,13 @@ func (k Keeper) createSigningRequest(
 			sdk.NewAttribute(types.AttributeKeyCurrentGroupSigningID, fmt.Sprintf("%d", currentGroupSigningID)),
 			sdk.NewAttribute(types.AttributeKeyIncomingGroupID, fmt.Sprintf("%d", incomingGroupID)),
 			sdk.NewAttribute(types.AttributeKeyIncomingGroupSigningID, fmt.Sprintf("%d", incomingGroupSigningID)),
+			sdk.NewAttribute(types.AttributeKeyTotalFee, totalFee.String()),
```
Comment (colmazia) [inline x/bandtss/keeper/keeper_signing.go:132]: Just to be clear so this means that if we do not enter this if if sender.String() != k.authority && currentGroupID != 0 { the total fee in the event will be zero. 
If this is intentional, you can resolve this.

File: x/bandtss/types/constructors.go [hunk 2]
```diff
@@ -58,11 +57,11 @@ func NewMember(
 // Validate performs basic validation of member information.
 func (m Member) Validate() error {
 	if _, err := sdk.AccAddressFromBech32(m.Address); err != nil {
-		return sdkerrors.ErrInvalidAddress.Wrapf("invalid member address: %s", err)
+		return ErrInvalidMember.Wrapf("invalid member address: %s", err)
 	}
 
 	if m.GroupID == 0 {
-		return ErrInvalidGroupID.Wrap("group id is 0")
+		return ErrInvalidMember.Wrap("group id is 0")
```
Comment (colmazia) [inline x/bandtss/types/constructors.go:64]: group id cannot be zero? dee mai

File: x/tss/types/constructors.go [hunk 3]
```diff
@@ -35,16 +35,40 @@ func NewGroup(
 
 // Validate performs basic validation of group information.
 func (g Group) Validate() error {
+	// validate group id
 	if g.ID == 0 {
 		return ErrInvalidGroup.Wrap("group id is 0")
```
Comment (colmazia) [inline x/tss/types/constructors.go:40]: ditto

File: x/tss/types/constructors.go [hunk 4]
```diff
@@ -35,16 +35,40 @@ func NewGroup(
...
+	if _, ok := GroupStatus_name[int32(g.Status)]; !ok {
+		return ErrInvalidGroup.Wrapf("invalid group status: %d", g.Status)
+	}
+	if g.Status == GROUP_STATUS_UNSPECIFIED {
+		return ErrInvalidGroup.Wrap("group status is unspecified")
+	}
+
+	// validate created height
+	if g.CreatedHeight == 0 {
+		return ErrInvalidGroup.Wrap("created height is 0")
```
Comment (colmazia) [inline x/tss/types/constructors.go:66]: ditto

File: x/tss/types/constructors.go [hunk 5]
```diff
@@ -35,16 +35,40 @@ func NewGroup(
...
+	}
+
+	// validate module owner
+	if g.ModuleOwner == "" {
+		return ErrInvalidGroup.Wrap("module owner is empty")
```
Comment (colmazia) [inline x/tss/types/constructors.go:71]: module owner cannot be empty

File: x/tss/types/constructors.go [hunk 6]
```diff
@@ -73,26 +97,31 @@ func NewRound1Info(
 
 // Validate performs basic validation of round-1 group creation information.
 func (r Round1Info) Validate() error {
+	// Validate member ID
+	if r.MemberID == 0 {
+		return ErrInvalidMember.Wrap("member id is 0")
```
Comment (colmazia) [inline x/tss/types/constructors.go:102]: ditto

File: x/tss/types/constructors.go [hunk 7]
```diff
@@ -121,7 +150,7 @@ func (r Round2Info) Validate() error {
 
 	for i, ess := range r.EncryptedSecretShares {
 		if err := ess.Validate(); err != nil {
-			return ErrInvalidSecretShare.Wrapf("encrypted secret shares at index %d: %s", i, err)
+			return ErrInvalidSecretShare.Wrapf("encrypted secret shares at index %d: %v", i, err)
```
Comment (colmazia) [inline x/tss/types/constructors.go:153]: fix "member id is 0" above too

File: x/tss/types/constructors.go [hunk 8]
```diff
@@ -162,23 +191,23 @@ func NewComplaint(
 // Validate performs basic validation of complaint information.
 func (c Complaint) Validate() error {
 	if c.Complainant == 0 {
-		return ErrInvalidMember.Wrap("complainant is 0")
+		return ErrInvalidComplaint.Wrap("complainant is 0")
 	}
 
 	if c.Respondent == 0 {
-		return ErrInvalidMember.Wrap("respondent is 0")
+		return ErrInvalidComplaint.Wrap("respondent is 0")
```
Comment (colmazia) [inline x/tss/types/constructors.go:198]: ditto

File: x/tss/types/member.go [hunk 9]
```diff
@@ -29,29 +28,24 @@ func NewMember(
 // Validate performs basic validation of group information.
 func (m Member) Validate() error {
 	if m.ID == 0 {
-		return ErrInvalidMember.Wrap("group id is 0")
+		return ErrInvalidMember.Wrap("member id is 0")
 	}
 
 	if m.GroupID == 0 {
-		return ErrInvalidMember.Wrap("group threshold is invalid")
+		return ErrInvalidMember.Wrap("group id is 0")
```
Comment (colmazia) [inline x/tss/types/member.go:35]: ditto

File: x/bandtss/keeper/keeper_signing.go (hunk 1 above)
Comment (nkitlabs) [reply x/bandtss/keeper/keeper_signing.go:132]: Yes, no fee is charged if not fall into those cases.

File: x/tss/types/constructors.go (hunk 7 above)
Comment (nkitlabs) [reply x/tss/types/constructors.go:153]: fixed

**Review by RogerKSI — APPROVED** [review]

**Review by colmazia — APPROVED** [review]

**Pull request merged**